tuf/repository_lib.py

```diff
@@ -1352,6 +1353,58 @@ def generate_targets_metadata(targets_directory, target_files, version,
 
 
 
+def _generate_targets_fileinfo(target_files, targets_directory,
+    write_consistent_targets):
+  """
+  Iterate over target_files and:
+    * ensure they exist in the targets_directory
+    * generate a fileinfo dict for the target file, including hashes
+    * copy 'target_path' to 'digest_target' if write_consistent_targets
+  add all generated fileinfo dicts to a dictionary mapping
+  targetpath: fileinfo and return the dict.
+  """
+
+  filedict = {}
+
+  # Generate the fileinfo of all the target files listed in 'target_files'.
+  for target, fileinfo in six.iteritems(target_files):
+
+    # The root-most folder of the targets directory should not be included in
+    # target paths listed in targets metadata.
+    # (e.g., 'targets/more_targets/somefile.txt' -> 'more_targets/somefile.txt')
+    relative_targetpath = target
+
+    # Note: join() discards 'targets_directory' if 'target' contains a leading
+    # path separator (i.e., is treated as an absolute path).
+    target_path = os.path.join(targets_directory, target.lstrip(os.sep))
+
+    # Ensure all target files listed in 'target_files' exist.  If just one of
+    # these files does not exist, raise an exception.
+    if not os.path.exists(target_path):
```

Comment:
Absolutely unrelated: This is the first of three `path.exists`/ `path.isfile` checks in the stack before we open that file (`get_metadata_fileinfo` has one and `securesystemslib.util.get_file_details` has one too). As good Pythonistas, we should ask for forgiveness instead.

Reply:
We should absolutely move towards the Python idiom of [EAFP](https://docs.python.org/3/glossary.html#term-eafp). I think we can do that in future changes as we work to address #1009